react/components/BuyButton/index.js

```diff
@@ -116,20 +126,36 @@ export const BuyButton = ({
         }
         const mutationRes = await orderFormContext.addItem({ variables })
         const { items } = mutationRes.data.addItem
+
         success = skuItems.filter(
           skuItem => !!items.find(({ id }) => id === skuItem.skuId)
         )
         await orderFormContext.refetch().catch(() => null)
       }
 
-      success =
-        success ||
-        (linkStateItems &&
-          skuItems.filter(
-            skuItem => !!linkStateItems.find(({ id }) => id === skuItem.skuId)
-          ))
+      const addedItem =
+        linkStateItems &&
+        skuItems.filter(
+          skuItem => !!linkStateItems.find(({ id }) => id === skuItem.skuId)
+        )
+
+      const foundItem =
+        orderFormItems &&
+        orderFormItems.filter(item => item.id === addedItem[0].skuId).length > 0
+
+      success = success || addedItem
 
-      showToastMessage = () => toastMessage(success && success.length >= 1)
+      showToastMessage = foundItem
```

Comment:
this should be: `() =>
            toastMessage({
              success: success && success.length >= 1,
              isNewItem: ! foundItem,
            })`

Reply:
Doone!